File: packages/android/plinth-android/src/main/java/io/plinth/android/PlinthConfig.kt
package io.plinth.android

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class PlinthConfig(
    val endpoint: String = "http://localhost:3000/beacon",
    @SerialName("project_key") val projectKey: String = "p123456789",
    @SerialName("heartbeat_interval_ms") val heartbeatIntervalMs: Long = 10_000L,
)

File: packages/apple/plinth-swift/Sources/PlinthSwift/PlinthConfig.swift
import Foundation

public struct PlinthConfig: Codable {
    public var endpoint: String
    public var projectKey: String
    public var heartbeatIntervalMs: UInt64

    public init(
        endpoint: String = "http://localhost:3000/beacon",
        projectKey: String = "p123456789",
        heartbeatIntervalMs: UInt64 = 10_000
    ) {
        self.endpoint = endpoint
        self.projectKey = projectKey
        self.heartbeatIntervalMs = heartbeatIntervalMs
    }

    public static let `default` = PlinthConfig()

    enum CodingKeys: String, CodingKey {
        case endpoint
        case projectKey = "project_key"
        case heartbeatIntervalMs = "heartbeat_interval_ms"
    }
}
